### ISSUE TITLE
No matching interaction found for GET...

### ISSUE THREAD

**Issue by carled** [pact-foundation/pact-net#262]
I'm pretty new to this and I'm obviously missing something fundamental. I went through a tutorial and successfully got pact running with the fake api. I then wanted to adapt it to run with my own api, but I cannot get it to recognise that I've created a mock service for the endpoint I'm using. I've tried with and without querystring parameters (my actual service doesn't use querystring, but the example I started with does so I'm just trying to get that working for now).

log file after debugging. No pact file is generated.

    I, [2020-10-10T12:12:49.484592 #34556]  INFO -- : Cleared interactions for example ""
    I, [2020-10-10T12:12:49.586595 #34556]  INFO -- : Registered expected interaction GET api/regions/?regionId=3
    D, [2020-10-10T12:12:49.586595 #34556] DEBUG -- : {
      "description": "A valid GET request for a region by Id",
      "providerState": "There is data",
      "request": {
        "method": "get",
        "path": "api/regions/",
        "query": "regionId=3"
      },
      "response": {
        "status": 200,
        "headers": {
          "Content-Type": "application/json; charset=utf-8"
        },
        "body": {
          "id": "3",
          "name": "Australia",
          "routingPlan": "Aus Customers",
          "onNetCallStaticContactUrl": "myurl.com",
          "inboundCallStaticContactUrl": "Hidden",
          "rtpProxy": "Direct",
          "diallingRuleId": 10189,
          "createdAt": "2020-01-03T10:48:33",
          "createdUser": "createdUser",
          "updatedUser": "updatedUser"
        }
      },
      "metadata": null
    }
    I, [2020-10-10T12:13:05.314825 #34556]  INFO -- : Received request GET /api/regions?regionId=3
    D, [2020-10-10T12:13:05.314825 #34556] DEBUG -- : {
      "path": "/api/regions",
      "query": "regionId=3",
      "method": "get",
      "headers": {
        "Host": "localhost:9222",
        "Version": "HTTP/1.1"
      }
    }
    E, [2020-10-10T12:13:05.314825 #34556] ERROR -- : No matching interaction found for GET /api/regions?regionId=3
    E, [2020-10-10T12:13:05.315826 #34556] ERROR -- : Interaction diffs for that route:
    E, [2020-10-10T12:13:05.315826 #34556] ERROR -- : 
    I, [2020-10-10T12:13:20.029341 #34556]  INFO -- : Cleared interactions for example ""

My consumer api client:

    static public async Task<HttpResponseMessage> GetRegionIdUsingProviderApi(string regionId, string baseUri)
        {
            using (var client = new HttpClient { BaseAddress = new Uri(baseUri) })
            {
                try
                {
                    var response = await client.GetAsync($"/api/regions?regionId={regionId}");
                    return response;
                }
                catch (System.Exception ex)
                {
                    throw new Exception("There was a problem connecting to Region Provider API.", ex);
                }
            }
        }

I know it'll be me doing something wrong, would just appreciate someone pointing out what I'm missing please!

**Comment by carled** [comment]
Just for comparison, in the same tests class I have the working example from the tutorial, which generates the following log:

    I, [2020-10-10T12:31:45.100577 #39160]  INFO -- : Cleared interactions for example ""
    I, [2020-10-10T12:31:45.179603 #39160]  INFO -- : Registered expected interaction GET /api/provider?validDateTime=04/05/2018
    D, [2020-10-10T12:31:45.180573 #39160] DEBUG -- : {
      "description": "A valid GET request for Date Validation",
      "providerState": "There is data",
      "request": {
        "method": "get",
        "path": "/api/provider",
        "query": "validDateTime=04/05/2018"
      },
      "response": {
        "status": 200,
        "headers": {
          "Content-Type": "application/json; charset=utf-8"
        },
        "body": {
          "test": "NO",
          "validDateTime": "04-05-2018 00:00:00"
        }
      },
      "metadata": null
    }
    I, [2020-10-10T12:31:50.830702 #39160]  INFO -- : Received request GET /api/provider?validDateTime=04/05/2018
    D, [2020-10-10T12:31:50.830702 #39160] DEBUG -- : {
      "path": "/api/provider",
      "query": "validDateTime=04/05/2018",
      "method": "get",
      "headers": {
        "Host": "localhost:9222",
        "Version": "HTTP/1.1"
      }
    }
    I, [2020-10-10T12:31:50.831701 #39160]  INFO -- : Found matching response for GET /api/provider?validDateTime=04/05/2018
    D, [2020-10-10T12:31:50.831701 #39160] DEBUG -- : {
      "status": 200,
      "headers": {
        "Content-Type": "application/json; charset=utf-8"
      },
      "body": {
        "test": "NO",
        "validDateTime": "04-05-2018 00:00:00"
      }
    }
    I, [2020-10-10T12:31:56.766742 #39160]  INFO -- : Writing pact for Provider_Region_API to ../../../../../pacts/consumer- provider_region_api.json
    I, [2020-10-10T12:31:56.782757 #39160]  INFO -- : Cleared interactions for example ""

And that hits this method in the consumerapiclient class:

        static public async Task<HttpResponseMessage> ValidateDateTimeUsingProviderApi(string dateTimeToValidate, string baseUri)
        {
            using (var client = new HttpClient { BaseAddress = new Uri(baseUri)})
            {
                try
                {
                    var response = await client.GetAsync($"/api/provider?validDateTime={dateTimeToValidate}");
                    return response;
                }
                catch (System.Exception ex)
                {
                    throw new Exception("There was a problem connecting to Provider API.", ex);
                }
            }
        }

**Comment by carled** [comment]
Well. This is weird. I spent ages with the working test and gradually changed its parameters and settings until it exactly matched the failing test, including pointing it at the same method in the ConsumerApiClient class. It worked. 

I then debugged the failing test. It also worked. 

I can only assume that some oddness was going on due to the oddity of the ruby mock server continuing to run if the test crashes out and when I stopped the running ruby.exe service, pact had a chance to recreate the interactions needed. I suspect this is the only possible reason - otherwise something really freaky is going on.

Anyhow - the above region Api test now works and it's still identical to the code as shown.
